### PR TITLE
[SKIP SOF-TEST] rebuild-testbench.sh: cosmetic improvements

### PR DESCRIPTION
**Pull request by marc-hb**
Minor cleanups spotted when looking at past issue 240cc9786ae245c

5 commits, please review separately.

The only really visible addition is this:

```
[100%] Linking C executable testbench
[100%] Built target testbench

Success!

For temporary, interactive Kconfiguration use:

   make -C /home/me/SOF/sof/tools/testbench/build_xt_testbench/sof_ep/build/  menuconfig

Permanent configuration is "src/arch/host/configs/library_defconfig".

For instant, incremental build:

   source  /home/me/SOF/sof/tools/testbench/build_xt_testbench/xtrun_env.sh
   make -C /home/me/SOF/sof/tools/testbench/build_xt_testbench/ -j8


....
```

### CODE REVIEW

File: scripts/rebuild-testbench.sh [hunk 1]
```diff
@@ -5,10 +5,15 @@
...
 
+SCRIPT_DIR=$(cd "$(dirname "$0")" && pwd)
+SOF_REPO=$(dirname "$SCRIPT_DIR")
+BUILD_TESTBENCH_DIR="$SOF_REPO"/tools/testbench
+
 # Defaults
 BUILD_TYPE=native
 BUILD_DIR_NAME=build_testbench
 BUILD_TARGET=install
+: "${SOF_AFL:=$HOME/sof/work/AFL/afl-gcc}"
```
Comment (singalsu) [inline scripts/rebuild-testbench.sh:17]: This is strange but it has been always in this script. Please add comment if you know what for it is.

**Review by singalsu — APPROVED** [review]
Thanks! Good updates, and all build types I use work correctly.

File: scripts/rebuild-testbench.sh (hunk 1 above)
Comment (marc-hb) [reply scripts/rebuild-testbench.sh:17]: This provides a value if none is found in the environment.

The syntax is cryptic but it's standard and documented, it's not even a bashism:
https://www.gnu.org/software/bash/manual/html_node/Shell-Parameter-Expansion.html
https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#tag_18_06_02

**Comment by marc-hb** [comment]
A couple git fetch timeouts in https://sof-ci.01.org/sof-pr-viewer/#/build/PR8611/build13263823, everything else green. That CI does not use the testbench at all.

**Review by lgirdwood — APPROVED** [review]

**Comment by lgirdwood** [comment]
@wszypelt @lrudyX good to merge ? I dont think this script is used by internal CI.

**Comment by wszypelt** [comment]
@lgirdwood Hello, recently there have been problems while building the build, the matter has been reported and I am looking into it, if you think that the changes do not apply to Internal Intel CI System, you can merge, in case I put PR back in the queue and I think that there should be results within an hour.

**Pull request merged**